third_party/move/move-compiler/transactional-tests/tests/control_flow/nested_for_while_loops.move

```diff
@@ -0,0 +1,20 @@
+//# run
+script {
+    fun main(): () {
+        let y = 0;
+        for (i in 0..10) {
+            y = y + 1;
+            while (y > 1) {
+                y = y - 1;
+            };
+        };
+        assert!(y == 1, 42);
+
+        while (y < 10) {
+            for (j in 0..2) {
```

Comment:
And a lot of incorrect implementations of this for loop will still leave y==11 at the end.

Reply:
Updated the test.